File: 2485 find-the-pivot-integer/solution.java
class Solution {
    public int pivotInteger(int n) {
        int sum=n*(n+1)/2;
        int lsum=0;
        for(int i=1;i<=n;i++) {
            if(lsum==sum-i-lsum) return i;
            lsum+=i;
        }
        return -1;
    }
}

File: 2485 find-the-pivot-integer/solution.rs
impl Solution {
    pub fn pivot_integer(n: i32) -> i32 {
        let sum = n*(n+1)/2;
        let mut lsum=0;
        for i in 1..=n {
            if lsum==sum-lsum-i {
                return i as i32
            }
            lsum+=i;
        }
        return -1
    }
}
